File: tuan02/bai38.jl
function da_sap_xep(a; nghiem_ngat=false)
    for i in 2:length(a)
        if nghiem_ngat
            if a[i] <= a[i - 1]
                return false
            end
        else
            if a[i] < a[i - 1]
                return false
            end
        end
    end
    return true
end

function main()
    line = readline()
    if isempty(strip(line)) return end
    
    tokens = split(line)
    n = parse(Int, tokens[1])
    a = [parse(Int, x) for x in tokens[2:end]]
    
    nn = da_sap_xep(a, nghiem_ngat=true)
    kg = da_sap_xep(a, nghiem_ngat=false)
    
    println("Nghiem ngat: ", nn ? "YES" : "NO")
    println("Khong giam: ", kg ? "YES" : "NO")
end

main()

File: tuan02/bai38.py
import sys

def da_sap_xep(a, nghiem_ngat=False) -> bool:
    for i in range(1, len(a)):
        if nghiem_ngat:
            if a[i] <= a[i - 1]:
                return False
        else:
            if a[i] < a[i - 1]:
                return False
    return True

def main():
    input_data = sys.stdin.read().split()
    if not input_data:
        return
    
    n = int(input_data[0])
    a = [int(x) for x in input_data[1:n + 1]]
    
    nn = da_sap_xep(a, nghiem_ngat=True)
    kg = da_sap_xep(a, nghiem_ngat=False)
    
    print(f"Nghiem ngat: {'YES' if nn else 'NO'}")
    print(f"Khong giam: {'YES' if kg else 'NO'}")

if __name__ == "__main__":
    main()